Microsoft.Toolkit.Uwp.UI.Controls/ImageCropper/ImageCropper.Logic.cs

```diff
@@ -0,0 +1,433 @@
+                UpdateSelectedRect(startPoint, endPoint);
+            }
+        }
+
+        /// <summary>
+        /// Update selection area.
+        /// </summary>
+        /// <param name="startPoint">The point on the upper left corner.</param>
+        /// <param name="endPoint">The point on the lower right corner.</param>
+        private void UpdateSelectedRect(Point startPoint, Point endPoint)
+        {
+            _startX = startPoint.X;
+            _startY = startPoint.Y;
+            _endX = endPoint.X;
+            _endY = endPoint.Y;
+            var centerX = ((_endX - _startX) / 2) + _startX;
+            var centerY = ((_endY - _startY) / 2) + _startY;
+            if (_topButton != null)
+            {
+                Canvas.SetLeft(_topButton, centerX);
+                Canvas.SetTop(_topButton, _startY);
+            }
+
+            if (_bottomButton != null)
+            {
+                Canvas.SetLeft(_bottomButton, centerX);
+                Canvas.SetTop(_bottomButton, _endY);
+            }
+
+            if (_leftButton != null)
+            {
+                Canvas.SetLeft(_leftButton, _startX);
+                Canvas.SetTop(_leftButton, centerY);
+            }
+
+            if (_rigthButton != null)
+            {
+                Canvas.SetLeft(_rigthButton, _endX);
+                Canvas.SetTop(_rigthButton, centerY);
+            }
+
+            if (_upperLeftButton != null)
+            {
+                Canvas.SetLeft(_upperLeftButton, _startX);
+                Canvas.SetTop(_upperLeftButton, _startY);
+            }
+
+            if (_upperRightButton != null)
+            {
+                Canvas.SetLeft(_upperRightButton, _endX);
+                Canvas.SetTop(_upperRightButton, _startY);
+            }
+
+            if (_lowerLeftButton != null)
+            {
+                Canvas.SetLeft(_lowerLeftButton, _startX);
+                Canvas.SetTop(_lowerLeftButton, _endY);
+            }
+
+            if (_lowerRigthButton != null)
+            {
+                Canvas.SetLeft(_lowerRigthButton, _endX);
+                Canvas.SetTop(_lowerRigthButton, _endY);
+            }
+
+            UpdateMaskArea();
+        }
+
+        /// <summary>
+        /// Update the mask layer.
+        /// </summary>
+        private void UpdateMaskArea()
+        {
+            if (_layoutGrid == null)
+            {
+                return;
+            }
+
+            _maskAreaGeometryGroup.Children.Clear();
+            _maskAreaGeometryGroup.Children.Add(new RectangleGeometry
+            {
+                Rect = new Rect(-_layoutGrid.Padding.Left, -_layoutGrid.Padding.Top, _layoutGrid.ActualWidth, _layoutGrid.ActualHeight)
+            });
+            if (CircularCrop)
+            {
+                var centerX = ((_endX - _startX) / 2) + _startX;
+                var centerY = ((_endY - _startY) / 2) + _startY;
+                _maskAreaGeometryGroup.Children.Add(new EllipseGeometry
```

Comment:
Is this the only thing needed for the crop, trying to trace where it masks the output bits.  Where is that done?  Wondering if using CornerRadius like the border controls would be possible/better than just square vs. circle.  Thoughts?

Reply:
I don’t think there are many use cases outside of square and rectangle.How about we keep it like this for now and add when there’s demand for other shapes?